lib/shelley/src/Cardano/Wallet/Shelley/Network/Blockfrost.hs

```diff
@@ -766,3 +1249,32 @@ getPoolPerformanceEstimate sp dl rp pid = do
             / fromIntegral (unCoin $ totalStake rp)
             -- _poolHistoryActiveSize would be incorrect here
         }
+
+newtype BFM a = BFM (ReaderT BF.ClientConfig (ExceptT BlockfrostError IO) a)
+    deriving newtype
+        ( Functor
+        , Applicative
+        , Monad
+        , MonadIO
+        , MonadBase IO
+        , MonadBaseControl IO
+        , MonadReader BF.ClientConfig
+        , MonadError BlockfrostError
+        )
+
+instance BF.MonadBlockfrost BFM where
+  getConf = ask
+  liftBlockfrostClient act = BFM $ do
+    env <- asks fst
+    liftIO (runClientM act env) >>=
+        either (throwError . ClientError . BF.fromServantClientError) pure
+
+newClientConfig :: BF.Project -> IO BF.ClientConfig
+newClientConfig prj = (, prj) <$> BF.newEnvByProject prj
+
+runBFM :: BF.ClientConfig -> BFM a -> IO a
+runBFM cfg (BFM c) = handleBlockfrostError (runReaderT c cfg)
```

Comment:
I forgot the details, I think that we had discusses this once: Do we reestablish the TLS connection every time that `runBFM` is called? If so, this would probably slow down `lightSync` considerably, given that we use `IO` as the monad there — but I realize that using `BFM` instead of `IO` is difficult to propagate through the rest of the code. 🙈 Maybe leave it for future work.

Reply:
The idea is that runBFM is ran once in the end, all blockfrost calls in it reuse the same connection pool.

Reply:
> Do we reestablish the TLS connection every time that runBFM is called?

I believe that we don't: `Network.HTTP.Client.Manager` is stored in the `BF.ClientConfig` and is supplied once when running BFM.